pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -60,20 +61,19 @@
 
   /**
    * Initializes the Segment Processor framework with input segments, output path and processing config
-   * @param inputSegmentsDir directory containing the input segments. These can be tarred or untarred.
+   * @param inputSegments list of input segments. These can be tarred or untarred.
    * @param segmentProcessorConfig config for segment processing
    * @param outputSegmentsDir directory for placing the resulting segments. This should already exist.
    */
-  public SegmentProcessorFramework(File inputSegmentsDir, SegmentProcessorConfig segmentProcessorConfig,
-      File outputSegmentsDir) {
+  public SegmentProcessorFramework(List<File> inputSegments, SegmentProcessorConfig segmentProcessorConfig,
+                                   File outputSegmentsDir) {
```

Comment:
Please follow the code style: https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup